src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
@hanslovsky Not sure how savvy you are about exceptions, but it is generally considered bad practice in my experience to throw `java.lang.Exception`. Too general, and a pain for consumers. You can throw a subclass of `RuntimeException` and it will be unchecked and then you can use `Function` as @dietzc suggests.